engine/engine.go

```diff
@@ -354,10 +355,10 @@ func (e *Engine) RegisterMultihashLister(mhl provider.MultihashLister) {
 // registered.
 //
 // See: Engine.RegisterMultihashLister, Engine.Publish.
-func (e *Engine) NotifyPut(ctx context.Context, contextID []byte, md metadata.Metadata) (cid.Cid, error) {
+func (e *Engine) NotifyPut(ctx context.Context, provider string, addrs []string, contextID []byte, md metadata.Metadata) (cid.Cid, error) {
```

Comment:
consider using `peer.AddrInfo` as the way to talk about the provider+addrs, and `peer.ID` rather than string when just referring to the provider's identity.

Reply:
Makes sense, will change